### PR TITLE
Adapt CI pipeline for multi-CassKop

### PR DESCRIPTION
**Pull request by erdrix**
Working on circle-ci pipeline, to adapt it to support multi-casskop

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -294,19 +401,19 @@ workflows:
...
           requires:
-            - unit-test
+            - unit-test-casskop
           filters:
             tags:
               only: /.*/
             branches:
               # Forked pull requests have CIRCLE_BRANCH set to pull/XXX
-              ignore: /pull\/[0-9]+/
+              ignore: /pull\/[0-9w]+/
```
Comment (cscetbon) [inline .circleci/config.yml]: @erdrix, not sure that's what you want here. Are you saying we have branches that have digits and `w`s only ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -366,10 +473,60 @@ workflows:
...
             tags:
               only: /.*/
             branches:
               # Forked pull requests have CIRCLE_BRANCH set to pull/XXX
               ignore: /pull\/[0-9]+/
+
+  # CassKop operator workflow.
+  ci-multicasskop-operator:
+    jobs:
+      - operator/generate:
```
Comment (cscetbon) [inline .circleci/config.yml]: @erdrix, those (generate) should not be jobs but steps

File: .circleci/config.yml [hunk 3]
```diff
@@ -82,207 +84,312 @@ orbs:
...
 
-      - unit-test:
+      - operator/build:
+          name: build-casskop
           requires:
-            - build
+            - unit-test-casskop
           filters:
             tags:
               only: /.*/
```
Comment (cscetbon) [inline .circleci/config.yml]: @erdrix, Our CI config is getting too long and complicated. Use [anchors](https://confluence.atlassian.com/bitbucket/yaml-anchors-960154027.html) as much as possible to refactor it and avoid repeating filters, steps etc...

File: multi-casskop/Makefile [hunk 4]
```diff
@@ -222,3 +231,8 @@ endif
 .PHONY: release
 release: tag image publish
 
+unit-test-with-vendor:
+	echo TODO
+#	$(UNIT_TEST_CMD_WITH_VENDOR) && echo "success!" || { echo "failure!"; cat test-report.out; exit 1; }
+#	cat test-report.out
+#	$(UNIT_TEST_COVERAGE)
```
Comment (cscetbon) [inline multi-casskop/Makefile]: Let's get rid of comments

File: .circleci/config.yml [hunk 5]
```diff
@@ -366,10 +473,60 @@ workflows:
...
+      - operator/generate:
+          name: generate-multicasskop
+          operatorDir: cassandra-k8s-operator/multi-casskop
+          operatorName: MultiCassKop
+          requires:
+            - generate-casskop
+          filters:
+            tags:
+              only: /.*/
+#      - operator/unit-test:
```
Comment (cscetbon) [inline .circleci/config.yml]: unused

**Comment by cscetbon** [comment]
@erdrix I know it's a draft but I started to review it just in case it helps

File: .circleci/config.yml (hunk 1 above)
Comment (erdrix) [reply .circleci/config.yml]: No, it was a mistake !

File: .circleci/config.yml (hunk 2 above)
Comment (erdrix) [reply .circleci/config.yml]: But something is not clear to me, how can we publish a docker image on the docker hub, before having played the unit-tests ?To me, it doesn't make sense, but maybe I'm missing something ?

File: .circleci/config.yml (hunk 3 above)
Comment (erdrix) [reply .circleci/config.yml]: Add some anchors to refactor yaml.

File: .circleci/config.yml (hunk 2 above)
Comment (cscetbon) [reply .circleci/config.yml]: It should be published but only as a temporary image or branch release to be used by all tests. Then only when the PR is merged the release (*-master) should be published

Comment (cscetbon) [reply .circleci/config.yml]: We should also have a job to publish a release when we tag it. Probably need a card for that ?

File: tools/publish_helm_gcs.sh [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+readonly GCS_BUCKET_INCUBATOR=gs://orange-kubernetes-charts-incubator
+
+main() {
+    mkdir -p tmp
+    setup_helm_client
+    authenticate
+
+#    if ! sync_repo stable "$GCS_BUCKET_STABLE" "$STABLE_REPO_URL"; then
+#        log_error "Not all stable charts could be packaged and synced!"
+#    fi
```
Comment (cscetbon) [inline tools/publish_helm_gcs.sh]: Any reason to still keep this ^ ?

Comment (erdrix) [reply tools/publish_helm_gcs.sh]: To uncomment it, once we will have a stable versions.

**Pull request merged**